pyroclast/common/visualizable.py

```diff
@@ -98,3 +98,25 @@ def sensitivity_map(self, x, softmax=False):
                 result = tf.nn.softmax(result)
             grads = tape.gradient(result, x)
         return grads
+
+    def smooth_grad(self, x, n=50, sigma=0.01):
+        """Computes the smooth grad saliency-map
+
+        SmoothGrad works by averaging the gradient of n input images
+        that have been perturbed with Gausssian noise. Gaussian noise
+        is parameterized by the standard deviation (sigma).
+
+        Args:
+           x (np.array): shape [batch_size, ...data_shape]
+           n (int): The number of samples to collect
+           sigma (float): The standard deviation of the Gaussian noise
+
+        Returns:
+           sensitivity_map (np.array): shape [batch_size, ...data_shape]
+
+        """
+        gradients = np.array(
```

Comment:
Why do this in numpy rather than TF?

Reply:
True. I wanted to use the old saliency map method, but it would probably be more efficient to do the whole thing in TF.

Reply:
It should be a straightforward change to update that for the merge.

Reply:
Just competed this.